redis_heartbeat_lock/async_lock.py

```diff
@@ -0,0 +1,120 @@
+"""Simple async wrapper around a Redis client to manage getting and holding a lock."""
+
+import asyncio
+import redis
+import time
+from typing import Any, Optional
+
+from .executor import run_sync_in_thread_pool
+
+
+class AsyncLock:
+    """An async wrapper around the officially supported Redis client for Python, used to implement basic locking."""
+
+    # The key to lock on
+    key: str
```

Comment:
Should these be underscore variables to mark them as not officially part of the public API?